cmd/config/subcommand/project/project_config.go

```diff
@@ -11,7 +18,78 @@ var (
 	}
 )
 
-// Config
+// Config holds the flag for get project
 type Config struct {
 	Filter filters.Filters `json:"filter" pflag:","`
 }
+
+//go:generate pflags ConfigProject --default-var DefaultProjectConfig --bind-default-var
+
+// ConfigProject hold configuration for project update flags.
+type ConfigProject struct {
+	ID              string            `json:"id" pflag:",id for the project specified as argument."`
+	ActivateProject bool              `json:"activateProject" pflag:",(Deprecated) Activates the project specified as argument. Only used in update"`
+	ArchiveProject  bool              `json:"archiveProject" pflag:",(Deprecated) Archives the project specified as argument. Only used in update"`
+	Activate        bool              `json:"activate" pflag:",Activates the project specified as argument. Only used in update"`
+	Archive         bool              `json:"archive" pflag:",Archives the project specified as argument. Only used in update"`
+	Name            string            `json:"name" pflag:",name for the project specified as argument."`
+	DryRun          bool              `json:"dryRun" pflag:",execute command without making any modifications."`
+	Description     string            `json:"description" pflag:",description for the project specified as argument."`
+	Labels          map[string]string `json:"labels" pflag:",labels for the project specified as argument."`
+	File            string            `json:"file" pflag:",file for the project definition."`
+}
+
+var DefaultProjectConfig = &ConfigProject{
+	Description: "",
+	Labels:      map[string]string{},
+}
+
+//GetProjectSpec return project spec from a file/flags
+func (c *ConfigProject) GetProjectSpec(id string) (*admin.Project, error) {
+	projectSpec := admin.Project{}
+	if len(c.File) > 0 {
+		yamlFile, err := ioutil.ReadFile(c.File)
+		if err != nil {
+			return nil, err
+		}
+		err = yaml.Unmarshal(yamlFile, &projectSpec)
+		if err != nil {
+			return nil, err
+		}
+		projectSpec.Id = id
+		return &projectSpec, nil
+	}
+
+	projectSpec.Id = id
+	projectSpec.Name = c.Name
+	projectSpec.Description = c.Description
+	projectSpec.Labels = &admin.Labels{
+		Values: c.Labels,
+	}
+	return &projectSpec, nil
+}
+
+//MapToAdminState return project spec from a file/flags
+func (c *ConfigProject) MapToAdminState(spec *admin.Project) (*admin.Project, error) {
```

Comment:
Just return state instead of project and you dont need to send the spec input, 
just use the returned value of the state in the consuming function to assign the returned state